File: Day36/RevVowelOfString.java
import java.util.Scanner;

public class RevVowelOfString {
    public String reverseVowels(String s) {
        String vowels = "aeiouAEIOU";
        char[] a = s.toCharArray();
        int left = 0, right = a.length - 1;
        while (left < right) {
            while (left < right && vowels.indexOf(a[left]) == -1) left++;
            while (left < right && vowels.indexOf(a[right]) == -1) right--;
            char temp = a[left];
            a[left] = a[right];
            a[right] = temp;
            left++;
            right--;
        }
        return new String(a);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String s = sc.nextLine();
        RevVowelOfString sol = new RevVowelOfString();
        System.out.println(sol.reverseVowels(s));
    }
}

File: Day36/RevVowelOfString.py
class Solution(object):
    def reverseVowels(self, s):
        vowels = 'aeiouAEIOU'
        a = list(s)
        b = []
        c = []
        for i in range(len(a)):
            if a[i] in vowels:
                b.append(a[i])
                c.append(i)
        b.reverse()
        for i in range(len(c)):
            a[c[i]] = b[i]
        return ''.join(a)

if __name__ == "__main__":
    s = input("Enter the string: ")
    print(Solution().reverseVowels(s))
